cmd/generate/config/rules/generic.go

```diff
@@ -35,7 +37,19 @@ func GenericCredential() *config.Rule {
 		Entropy: 3.5,
 		Allowlists: []config.Allowlist{
 			{
-				StopWords: DefaultStopWords,
+				Description:    "Allowlist for Generic API Keys",
+				MatchCondition: config.AllowlistMatchOr,
+				Regexes: []*regexp.Regexp{
+					regexp.MustCompile(`(?i)(` +
+						`public[_.-]?(key|token)` + // public key -> not a secret
+						`|(api|client)[_.-]?(version|id)` + // version/id -> not a secret
+						`|(secret|client)[_.-]?name` + // name of client, name of env variable, ...
+						`|key[_.-]?word|monkey|donkey|keyboard|keying|keystone` + // common words containing "key"
```

Comment:
Is it possible that `monkey` and `donkey` could inadvertently match something like `...monKey`? Perhaps we should require a specific case.

```suggestion
						`|key[_.-]?word|(?-i:[DdMm]onkey|[DM]ONKEY)|keyboard|keying|keystone` + // common words containing "key"
```

Reply:
Good point. And I was not even aware you could switch case-invariant off again.

Reply:
I am wondering, whether it would be better to limit the keywords:
instead of matching case-invariant for "key", only match for 
- "Key"
- "key", not preceded or followed by lowercase letter, 
- "KEY", not preceded or followed by uppercase letter
something like this:
```regex
(?-i:(?:Key[^a-z])|(?:(?:\b|[^a-z\s])key[^a-z])|(?:(?:\b|[^A-Z\s])KEY[^A-Z]))
```
Then monkey, donkey would automatically be excluded. But it seems a little convoluted.

Reply:
> ```regexp
> (?-i:(?:Key[^a-z])|(?:(?:\b|[^a-z\s])key[^a-z])|(?:(?:\b|[^A-Z\s])KEY[^A-Z]))
> ```
> 
> Then monkey, donkey would automatically be excluded. But it seems a little convoluted.

That is a clever pattern; unfortunately, I think this has to be explicitly tweaked on a case-by-case basis.

Otherwise, there's the risk of false negatives such as:
```yaml
snyk:
  apikey: 04def1de-680f-4340-bdad-78fabf18380f
```